fetch/fetch.go

```diff
@@ -128,26 +141,31 @@ func NewMessageNetwork(requestTimeOut int, net service.Service, protocol string,
 }
 
 // GetRandomPeer returns a random peer from current peer list
-func (f MessageNetwork) GetRandomPeer() p2ppeers.Peer {
-	peers := f.peersProvider.GetPeers()
+func GetRandomPeer(peers []p2ppeers.Peer) p2ppeers.Peer {
 	if len(peers) == 0 {
-		f.Log.Panic("cannot send fetch - no peers found")
+		log.Panic("cannot send fetch - no peers found")
```

Comment:
should we really panic here?

Reply:
what should we do if we try to get a peer and dont have one?